### ISSUE TITLE
Questions? Feedback?

### ISSUE THREAD

**Issue by andywer** [andywer/pg-lint#1]
Add your feedback as a comment to this issue 📋

**Comment by jtremback** [comment]
Just wanted to say that this looks awesome. Not sure we will use it given its early stage though.

**Comment by hassanNS** [comment]
Hi, this project looks amazing. Is it possible to use this for validating plain MYSQL?

**Comment by andywer** [comment]
> Is it possible to use this for validating plain SQL and not postgres sql?

Since Postgres' SQL dialect is just a superset of "vanilla" SQL, there should be no problem validating that 👍

I guess the only risk would be to use something too fancy that postgres/postguard understand and accept, but your non-postgres database (sqlite for instance) does not.

**Comment by gregplaysguitar** [comment]
This looks super cool. The thing that would make it perfect, from my point of view, would be if you could point it at an actual postgres schema definition (i.e. a directory containing .sql files) as the source of truth for the database, rather than defining it with `defineTable`. How much of a leap would that be?

**Comment by andywer** [comment]
@gregplaysguitar Yeah, that would be great! Shouldn't be too hard either, now as I think about it…

Just not sure if I will have time to work on it anytime soon. I've been sick for the last week and a half and stuff is already piling up 😉

**Comment by gregplaysguitar** [comment]
@andywer do you have any pointers as to where to start with that? I'd love to help out, although I don't have much free time either...

**Comment by andywer** [comment]
@gregplaysguitar You can have a look at:
- [`parseTableDefinition()` in `src/babel/parse-table-definition.ts`](https://github.com/andywer/postguard/blob/master/src/babel/parse-table-definition.ts): It takes a piece of source code parsed by Babel and returns a `TableSchema`
- [`src/postgres/parse-pg-query.ts`](https://github.com/andywer/postguard/blob/master/src/postgres/parse-pg-query.ts): Here you can see how SQL queries are parsed using `pg-query-parser` (not super intuitive)

I think the best way to get started with the `pg-query-parser` package is to write a tiny test script that parses a simple SQL query and prints the parsed AST. Then play around with the input query a little bit and check out the resulting output.

**Comment by gregplaysguitar** [comment]
Great, thanks! I’ve been wanting to dig into pg query parsing for another project too so keen to have a look at this. I’ll let you know how I go!

**Comment by LifeIsStrange** [comment]
Can it validate queries at runtime?

**Comment by andywer** [comment]
No, it's build-time-only 😉 @LifeIsStrange

**Comment by LifeIsStrange** [comment]
that's unfortunate :)
I have a use case where users write SQL queries on a web frontend and it would be nice to check the syntax validity on the client instead of doing roundtrips to the server.
The only solution I've found is https://github.com/paxdei/GWTJSqlParser but it is a bit obscure haha
